rasa/nlu/tokenizers/whitespace_tokenizer.py

```diff
@@ -1,42 +1,82 @@
+from rasa.engine.storage.storage import ModelStorage
+from rasa.nlu.tokenizers.tokenizer import Token, TokenizerGraphComponent
 from rasa.shared.constants import DOCS_URL_COMPONENTS
-from rasa.nlu.tokenizers.tokenizer import Token, Tokenizer
 from rasa.shared.nlu.training_data.message import Message
 
+from rasa.nlu.tokenizers._whitespace_tokenizer import WhitespaceTokenizer
+
 
-class WhitespaceTokenizer(Tokenizer):
+# This is a workaround around until we have all components migrated to `GraphComponent`.
+WhitespaceTokenizer = WhitespaceTokenizer
 
-    defaults = {
-        # Flag to check whether to split intents
-        "intent_tokenization_flag": False,
-        # Symbol on which intent should be split
-        "intent_split_symbol": "_",
-        # Regular expression to detect tokens
-        "token_pattern": None,
-    }
 
-    # the following language should not be tokenized using the WhitespaceTokenizer
-    not_supported_language_list = ["zh", "ja", "th"]
+class WhitespaceTokenizerGraphComponent(TokenizerGraphComponent):
+    @staticmethod
+    def not_supported_languages() -> Optional[List[Text]]:
+        """The languages that are not supported (see parent class for full docstring)."""
+        return ["zh", "ja", "th"]
 
-    def __init__(self, component_config: Dict[Text, Any] = None) -> None:
-        """Construct a new tokenizer using the WhitespaceTokenizer framework."""
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        return {
+            # Flag to check whether to split intents
+            "intent_tokenization_flag": False,
+            # Symbol on which intent should be split
+            "intent_split_symbol": "_",
+            # Regular expression to detect tokens
+            "token_pattern": None,
+        }
 
-        super().__init__(component_config)
+    def __init__(
+        self, config: Dict[Text, Any], model_storage: ModelStorage, resource: Resource,
+    ) -> None:
+        """Initialize the tokenizer."""
+        super().__init__(config)
 
+        self._model_storage = model_storage
+        self._resource = resource
         self.emoji_pattern = rasa.utils.io.get_emoji_regex()
 
-        if "case_sensitive" in self.component_config:
+        if "case_sensitive" in self._config:
             rasa.shared.utils.io.raise_warning(
                 "The option 'case_sensitive' was moved from the tokenizers to the "
                 "featurizers.",
                 docs=DOCS_URL_COMPONENTS,
             )
 
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> "WhitespaceTokenizerGraphComponent":
+        """Creates a new component (see parent class for full docstring)."""
+        # Path to the dictionaries on the local filesystem.
+        return cls(config, model_storage, resource)
+
+    @classmethod
+    def load(
```

Comment:
We shouldn't need this method as the default in `GraphComponent` just calls `create`.